init/arch_specs/eessi_arch_x86.spec

```diff
@@ -0,0 +1,6 @@
+# x86_64 CPU architecture specifications
+# Software path in EESSI 	| Vendor ID 	| List of defining CPU features
+"x86_64/intel/haswell"		"GenuineIntel"	"avx2 fma" 		# Intel Haswell, Broadwell
+"x86_64/intel/skylake_avx512"	"GenuineIntel"	"avx2 fma avx512f"	# Intel Skylake, Cascade Lake
```

Comment:
If I read this correctly this could be misidentified as Xeon Phi Knights Landing.
Skylake and Cascade Lake have `avx512bw avx512vl avx512dq`, which Knights Landing doesn't have.

Reply:
the conscious idea was to look for the -minimal- required cpu features to have EESSI set of binaries run optimally, in contrast to archspec which tries take the (over?) complete set of features. The current combination of features is just a hopefully sane start, which work for the current set of cpu's in the test scenarios. Please do provide a cpuinfo from KL so we can add it. Also let's test verify that the current codes indeed do not run on KL... Thanks!

Reply:
I no longer have access to a KNL node but https://colfaxresearch.com/knl-avx512/ has this:
```
vega@lyra% cat /proc/cpuinfo
...
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cm
ov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdt
scp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aper
fmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 fma cx16 xtp
r pdcm sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdr
and lahf_lm abm 3dnowprefetch ida arat epb xsaveopt pln pts dtherm tpr_shadow vnmi 
flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms avx512f rdseed 
adx avx512pf avx512er avx512cd
...```

Knights Landing is getting pretty rare so there is no reason to compile with only support for `avx512f`; excluding bw/vl/dq isn't good for performance. It just shouldn't misidentify. Haswell `avx2` code will run fine on KNL, just without optimal performance.

Reply:
skylake_avx512 updated to include '''AVX512F, AVX512BW, AVX512CD, AVX512DQ, AVX512VL'''